cherrypy/test/test_httputil.py

```diff
@@ -79,3 +80,84 @@ def test_invalid_status(status_code, error_msg):
     """Check that invalid status cause certain errors."""
     with pytest.raises(ValueError, match=error_msg):
         httputil.valid_status(status_code)
+
+
+@pytest.mark.parametrize(
+    ('header_content', 'value', 'params'),
+    [
+        ('application/x-www-form-urlencoded',
+         'application/x-www-form-urlencoded', {}),  # default_content_type
+        ('application/json;charset="utf8"',
+         'application/json', {'charset': 'utf8'}),
+        ('audio/*; q=0.2, audio/basic',
+         'audio/*', {'q': '0.2, audio/basic'}),
+        ('text/plain; q=0.5, text/html, text/x-dvi; q=0.8, text/x-c',
+         'text/plain', {'q': '0.8, text/x-c'}),
+        ('text/*, text/html, text/html;level=1, */*',
+         'text/*, text/html, text/html', {'level': '1, */*'}),
+        ('iso-8859-5, unicode-1-1;q=0.8',
+         'iso-8859-5, unicode-1-1', {'q': '0.8'}),
+        ('gzip;q=1.0, identity; q=0.5, *;q=0',
+         'gzip', {'q': '0'}),
+        ('da, en-gb;q=0.8, en;q=0.7',
+         'da, en-gb', {'q': '0.7'}),
+        ('text/plain',
+         'text/plain', {}),
+        ('text/xml',
+         'text/xml', {}),
+    ]
+)
+def test_header_element(header_content, value, params):
+    """ Test that value and params are parsed from headers.
+
+        Testing that the value and params are parsed from
+        headers passed to HeaderElement from httputils """
+    hdr_elem = httputil.HeaderElement.from_str(header_content)
+
+    assert hdr_elem.value == value
+    assert hdr_elem.params == params
+
+
+@pytest.mark.parametrize(
+    'header_content,media_type,qvalue',
+    [
+        ('application/x-www-form-urlencoded',
+         'application/x-www-form-urlencoded', 1.0),  # default_content_type
+        ('application/json;charset="utf8"',
+         'application/json', 1.0),
+        ('text/*, text/html, text/html;level=1, */*',
+         'text/*, text/html, text/html', 1.0),
+        ('iso-8859-5, unicode-1-1;q=0.8',
+         'iso-8859-5, unicode-1-1', 0.8),
+        ('text/plain',
+         'text/plain', 1.0),
+        ('text/xml',
+         'text/xml', 1.0),
+    ]
+)
+def test_accept_element(header_content, media_type, qvalue):
+    """ Test that value and qvalue are parsed from headers passed to
+        AcceptElement from httputils """
+    acc_elem = httputil.AcceptElement.from_str(header_content)
+    assert acc_elem.value == media_type
+    assert acc_elem.qvalue == qvalue
+
+
+@pytest.mark.parametrize(
+    'header_content,media_type,qvalue',
```

Comment:
I thought we had a linter for this... Let's enumerate the params list as an iterable of strings:
```suggestion
    ('header_content', 'media_type', 'qvalue'),
```